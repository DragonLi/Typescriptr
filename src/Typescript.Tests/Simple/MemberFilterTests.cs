using System;
using System.Reflection;
using Assent;
using Typescriptr;
using Xunit;

namespace Typescript.Tests.Simple
{
    public class MemberFilterTests
    {
        private class IgnoreAttribute : Attribute {}
        
        private class SimpleClass
        {
            public int SerializedType { get; set; }
            [Ignore] public int IgnoredType { get; set; }
        }

        [Fact]
        public void WhenNoFilterSpecified_TypeIsEmitted()
        {
            var generator = TypeScriptGenerator.CreateDefault();
            var generated = generator.Generate(new[] {typeof(SimpleClass)});
            
            this.Assent(generated.JoinTypesAndEnums());
        }

        [Fact]
        public void WhenFilterSpecified_PropertyIsIgnored()
        {
            var generator = TypeScriptGenerator
                    .CreateDefault()
                    .WithMemberFilter(memberInfo => memberInfo.GetCustomAttribute<IgnoreAttribute>() == null)
                ;
            var generated = generator.Generate(new[] {typeof(SimpleClass)});
            
            this.Assent(generated.JoinTypesAndEnums());
        }
    }
}