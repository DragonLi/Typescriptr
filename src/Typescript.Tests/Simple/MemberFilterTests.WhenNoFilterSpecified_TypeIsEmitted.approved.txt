declare namespace Api {
  interface SimpleClass {
    serializedType: number;
    ignoredType: number;
  }
}

---
